### PR TITLE
Add seo settings for download/extern link and target settings for download link

### PR DESCRIPTION
**Pull request by VaniW**

### CODE REVIEW

File: Kwc/Basic/DownloadTag/Component.php [hunk 1]
```diff
@@ -23,10 +25,22 @@ public function getTemplateVars(Kwf_Component_Renderer_Abstract $renderer)
         $row = $this->_getRow();
         $filename = $row->filename != '' ? $row->filename : 'unnamed';
 
-
         $ret['filesize'] = $this->getFilesize();
         $ret['url'] = $this->getDownloadUrl();
         $ret['filename'] = $filename;
+
+        $rel = $this->getData()->rel ? $this->getData()->rel : array();
```
Comment (kaufmo) [inline Kwc/Basic/DownloadTag/Component.php:32]:
```suggestion
        $rel = $this->getData()->rel ? array($this->getData()->rel) : array();
```

File: Kwc/Basic/LinkTag/Extern/Component.php [hunk 2]
```diff
@@ -17,4 +17,24 @@ public static function getSettings($param = null)
         return $ret;
     }
 
+    public function getTemplateVars(Kwf_Component_Renderer_Abstract $renderer)
+    {
+        $ret = parent::getTemplateVars($renderer);
+
+        $row = $this->_getRow();
+        $rel = $this->getData()->rel ? $this->getData()->rel : array();
```
Comment (kaufmo) [inline Kwc/Basic/LinkTag/Extern/Component.php:25]:
```suggestion
        $rel = $this->getData()->rel ? array($this->getData()->rel) : array();
```

File: Kwc/Basic/LinkTag/Extern/Component.php [hunk 3]
```diff
@@ -17,4 +17,24 @@ public static function getSettings($param = null)
...
+        if ($row->rel_nofollow) {
+            $rel[] = 'nofollow';
+        }
+        if ($row->rel_noopener) {
+            $rel[] = 'noopener';
+        }
+        if ($row->rel_noreferrer) {
+            $rel[] = 'noreferrer';
+        }
+        $ret['rel'] = implode(' ', array_unique($rel));
```
Comment (kaufmo) [inline Kwc/Basic/LinkTag/Extern/Component.php:35]: is this rel used? in abstract component tpl there is only $this->data->rel used not $this->rel?

File: Kwc/Basic/LinkTag/Extern/Component.tpl [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?php
+if ($this->data->url) {
+    echo '<a class="'.$this->linkClass.'" href="' . Kwf_Util_HtmlSpecialChars::filter($this->data->url) . '"';
+    if($this->rel) { echo ' rel="' . Kwf_Util_HtmlSpecialChars::filter($this->rel) . '"'; }
```
Comment (kaufmo) [inline Kwc/Basic/LinkTag/Extern/Component.tpl:4]:
```suggestion
    if ($this->rel) { echo ' rel="' . Kwf_Util_HtmlSpecialChars::filter($this->rel) . '"'; }
```

File: Kwc/Basic/DownloadTag/Component.php [hunk 5]
```diff
@@ -23,10 +25,22 @@ public function getTemplateVars(Kwf_Component_Renderer_Abstract $renderer)
         $row = $this->_getRow();
         $filename = $row->filename != '' ? $row->filename : 'unnamed';
 
-
         $ret['filesize'] = $this->getFilesize();
         $ret['url'] = $this->getDownloadUrl();
         $ret['filename'] = $filename;
+
+        $rel = $this->getData()->rel ? array($this->getData()->rel) : array();
```
Comment (kaufmo) [inline Kwc/Basic/DownloadTag/Component.php:32]: that's not needed anymore?? code is in data.php?

File: Kwc/Basic/DownloadTag/Data.php [hunk 6]
```diff
@@ -13,25 +13,75 @@ public function __get($var)
...
             }
-            $filename .= '.'.$fRow->extension;
+            $filename .= '.' . $fRow->extension;
             $ret = Kwf_Media::getUrl($this->componentClass, $this->componentId, 'default', $filename);
             $ev = new Kwf_Component_Event_CreateMediaUrl($this->componentClass, $this, $ret);
             Kwf_Events_Dispatcher::fireEvent($ev);
             return $ev->url;
+        } else if ($var == 'rel') {
+            $rel = parent::__get($var);
+            $rel = $rel ? array($rel) : array();
```
Comment (kaufmo) [inline Kwc/Basic/DownloadTag/Data.php:23]:
```suggestion
            $rel = array(parent::__get($var));
```

**Review by Ben-Ho — APPROVED** [review]

**Review by fraxachun — APPROVED** [review]

**Pull request merged**